sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -260,32 +268,63 @@ def _delay_failure(start_time: datetime.datetime) -> None:
-        correlation_context = "RequestType=" + request_type
-        if (
-            "keyvault_credential" in kwargs
-            or "keyvault_client_configs" in kwargs
-            or "secret_resolver" in kwargs
-            or kwargs.pop("uses_key_vault", False)
-        ):
-            correlation_context += ",UsesKeyVault"
-        host_type = ""
-        if AzureFunctionEnvironmentVariable in os.environ:
-            host_type = "AzureFunction"
-        elif AzureWebAppEnvironmentVariable in os.environ:
-            host_type = "AzureWebApp"
-        elif ContainerAppEnvironmentVariable in os.environ:
-            host_type = "ContainerApp"
-        elif KubernetesEnvironmentVariable in os.environ:
-            host_type = "Kubernetes"
-        elif ServiceFabricEnvironmentVariable in os.environ:
-            host_type = "ServiceFabric"
-        if host_type:
-            correlation_context += ",Host=" + host_type
-        headers["Correlation-Context"] = correlation_context
+    if os.environ.get(REQUEST_TRACING_DISABLED_ENVIRONMENT_VARIABLE, default="").lower() == "true":
+        return headers
+    correlation_context = "RequestType=" + request_type
+
+    if "feature_filters_used" in kwargs:
+        filters_used = ""
+        feature_filters_used = kwargs.pop("feature_filters_used", {})
+        if CUSTOM_FILTER_KEY in feature_filters_used:
+            filters_used = CUSTOM_FILTER_KEY
+        if PERCENTAGE_FILTER_KEY in feature_filters_used:
+            filters_used += ("+" if filters_used else "") + PERCENTAGE_FILTER_KEY
+        if TIME_WINDOW_FILTER_KEY in feature_filters_used:
+            filters_used += ("+" if filters_used else "") + TIME_WINDOW_FILTER_KEY
+        if TARGETING_FILTER_KEY in feature_filters_used:
+            filters_used += ("+" if filters_used else "") + TARGETING_FILTER_KEY
+        correlation_context += ",Filters=" + filters_used
+
+    correlation_context += _uses_feature_flags(**kwargs)
+
+    if (
+        "keyvault_credential" in kwargs
+        or "keyvault_client_configs" in kwargs
+        or "secret_resolver" in kwargs
+        or kwargs.pop("uses_key_vault", False)
+    ):
+        correlation_context += ",UsesKeyVault"
+    host_type = ""
+    if AzureFunctionEnvironmentVariable in os.environ:
+        host_type = "AzureFunction"
+    elif AzureWebAppEnvironmentVariable in os.environ:
+        host_type = "AzureWebApp"
+    elif ContainerAppEnvironmentVariable in os.environ:
+        host_type = "ContainerApp"
+    elif KubernetesEnvironmentVariable in os.environ:
+        host_type = "Kubernetes"
+    elif ServiceFabricEnvironmentVariable in os.environ:
+        host_type = "ServiceFabric"
+    if host_type:
+        correlation_context += ",Host=" + host_type
+
+    headers["Correlation-Context"] = correlation_context
     return headers
 
 
+def _uses_feature_flags(**kwargs):
+    if "uses_feature_flags" not in kwargs or not kwargs.pop("uses_feature_flags", False):
```

Comment:
uses_feature_flags=kwargs.pop("uses_feature_flags", False)
if not uses_feature_flags:
     return ""

?

Reply:
I guess the None check isn't needed with the default value.

The idea here is that we only send back the telemetry that it's used if:
1. The library is installed
2. We are loading feature flags from App Configuration